src/Microsoft.Azure.WebJobs.Host/Scale/ConcurrencyManager.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Used to implement collaborative dynamic concurrency management between the host and function triggers.
+    /// Function listeners can call <see cref="GetStatus"/> within their listener polling loops to determine the
+    /// amount of new work that can be fetched. The manager internally adjusts concurrency based on various
+    /// health heuristics.
+    /// </summary>
+    public class ConcurrencyManager
```

Comment:
Still debating whether this should be an IConcurrencyManager interface rather than concrete type. I left it a type since this is the focal point for DC, and I want the flexibility to add new functionality w/o breaking interface changes.

Reply:
This makes sense to me.